internal/reporter.go

```diff
@@ -17,15 +18,26 @@ type reporter struct {
 	client    *http.Client
 }
 
-// NewReporter create a metrics Reporter
-func NewReporter(server string, token string) Reporter {
+// NewReporter creates a metrics Reporter
+func NewReporter(server string, token string, client *http.Client) Reporter {
 	return &reporter{
 		serverURL: server,
 		token:     token,
-		client:    &http.Client{Timeout: time.Second * 10},
+		client:    client,
 	}
 }
 
+func NewClient(timeout time.Duration, tlsConfig *tls.Config) *http.Client {
+	var client *http.Client
+	if tlsConfig == nil {
+		client = &http.Client{Timeout: timeout}
```

Comment:
You could simplify this function like so:
```
if tlsConfig == nil {
    return &http.Client{Timeout: timeout}
}
transport := &http.Transport{TLSClientConfig: tlsConfig}
return &http.Client{Timeout: timeout, Transport: transport}
```

Reply:
I'm confused by the change as-is @suprajanarasimhan:
```go
	if tlsConfig == nil {
		client = &http.Client{Timeout: timeout}
	}		
	transport := &http.Transport{TLSClientConfig: tlsConfig}
	client = &http.Client{Timeout: timeout, Transport: transport}

    return client
```
The `if` condition will have no effect on `client` since it's overwritten below that. Did you mean to follow Travis' original suggestion?

```go
if tlsConfig == nil {
    return &http.Client{Timeout: timeout}
}
transport := &http.Transport{TLSClientConfig: tlsConfig}
return &http.Client{Timeout: timeout, Transport: transport}
```

Reply:
@suprajanarasimhan, can we add test coverage with a nil and non-nil `tlsConfig`, to catch this logic error?

Reply:
As Glenn already stated, consider adding a test for internal.NewClient().  When the tlsConfig parameter is nil, the Transport field of returned http client should be nil.

Reply:
Hi @oppegard, thank for catching this logic error! 🙏🏽 @ktollivervmw, @jessepye and I added the test you recommended.